python/ray/rllib/optimizers/multi_gpu.py

```diff
@@ -52,33 +51,37 @@ def _init(self, sgd_batch_size=128, sgd_stepsize=5e-5, num_sgd_iter=10,
         print("LocalMultiGPUOptimizer batch size", self.batch_size)
 
         # List of (feature name, feature placeholder) tuples
-        self.loss_inputs = self.local_evaluator.tf_loss_inputs()
+        self.loss_inputs = self.local_evaluator.for_policy(lambda p: p.loss_in)
 
         # per-GPU graph copies created below must share vars with the policy
-        main_thread_scope = tf.get_variable_scope()
         # reuse is set to AUTO_REUSE because Adam nodes are created after
         # all of the device copies are created.
-        with tf.variable_scope(main_thread_scope, reuse=tf.AUTO_REUSE):
-            self.par_opt = LocalSyncParallelOptimizer(
-                tf.train.AdamOptimizer(self.sgd_stepsize),
-                self.devices,
-                [ph for _, ph in self.loss_inputs],
-                self.per_device_batch_size,
-                lambda *ph: self.local_evaluator.build_tf_loss(ph),
-                os.getcwd())
-
-        # TODO(rliaw): Find more elegant solution for this
-        if hasattr(self.local_evaluator, "init_extra_ops"):
-            self.local_evaluator.init_extra_ops(
-                self.par_opt.get_device_losses())
-
-        self.sess = self.local_evaluator.sess
-        self.sess.run(tf.global_variables_initializer())
+        with self.local_evaluator.sess.graph.as_default():
+            main_thread_scope = tf.get_variable_scope()
+            with tf.variable_scope(main_thread_scope, reuse=tf.AUTO_REUSE):
+                def build_loss(inputs):
+                    cfg = self.local_evaluator.policy_config
+                    env = self.local_evaluator.env
+                    ac_space = env.get_unwrapped().action_space
+                    return self.local_evaluator.policy_graph(
```

Comment:
Instead of invoking a custom constructor, how about we add a `.copy(new_inputs)` method on TFPolicyGraph? The contract can be that this method needs to be implemented for multi gpu optimizer to work.